util/testutil/shell.go

```diff
@@ -141,6 +142,42 @@ func (m *RemoteSnapshotMonitor) CheckAllRemoteSnapshots(t *testing.T) {
 	}
 }
 
+// FatalMonitor scans logs streamed from the specified io.Reader for fatal errors during startup
+func FatalMonitor(r shell.Reporter, stdout, stderr io.Reader) {
+	errs := make(chan error, 1)
+	go ScanFatalLog(io.TeeReader(stdout, r.Stdout()), errs)
+	go ScanFatalLog(io.TeeReader(stderr, r.Stderr()), errs)
+	err := <-errs
+	if err != nil {
+		r.Errorf(err.Error())
+	}
+}
+
+type LevelLogLine struct {
+	Level string `json:"level"`
+	Msg   string `json:"msg"`
+}
+
+// ScanFatalLog scans the log streamed from the specified io.Reader for fatal errors during startup
+func ScanFatalLog(inputR io.Reader, errs chan error) {
+	scanner := bufio.NewScanner(inputR)
+	var logline LevelLogLine
+	for scanner.Scan() {
+		rawL := scanner.Text()
+		if i := strings.Index(rawL, "{"); i > 0 {
+			rawL = rawL[i:] // trim garbage chars; expects "{...}"-styled JSON log
+		}
+		if err := json.Unmarshal([]byte(rawL), &logline); err == nil {
+			if logline.Level == "fatal" {
+				errs <- errors.New("fatal snapshotter log entry encountered")
```

Comment:
```suggestion
				errs <- errors.New("aborting test: fatal snapshotter log entry encountered")
```

Reply:
Or maybe this should stay as it is and `FatalMonitor` should wrap the error with "aborting test" before sending the error to the reporter?